src/bin/polymer-bundler.ts

```diff
@@ -224,59 +228,94 @@ if (options.redirect) {
             return {prefix, path};
           })
           .filter((r: redirection) => r.prefix && r.path);
-  const resolvers: UrlResolver[] =
-      redirections.map((r: redirection) => new RedirectionResolver(r.prefix));
+  const resolvers: UrlResolver[] = redirections.map((r: redirection) => {
+    let resolvedPath = Uri.file(path.resolve(r.path)).toString();
+    // Ensure trailing slash if input path had trailing slash
+    if (r.path.match(/\\$|\/$/)) {
+      resolvedPath = ensureTrailingSlash(resolvedPath);
+    }
+    return new RedirectResolver(
+        packageUrlResolver.resolve(r.prefix as PackageRelativeUrl)!,
+        r.prefix,
+        resolvedPath);
+  });
+
+  // Wrap the getPackageRelativeUrl function for each redirection.  Because we
+  // are wrapping the function, we iterate in reverse order to ensure the
+  // redirect transformations are processed in the same order as they're given.
+  redirections.reverse().forEach((r: redirection) => {
+    const oldGetPackageRelativeUrl = getPackageRelativeUrl;
+    const newGetPackageRelativeUrl = function(resolvedUrl: ResolvedUrl):
+        PackageRelativeUrl {
+          const redirectionPathUrl =
+              ensureTrailingSlash(Uri.file(path.resolve(r.path)).toString());
+          if (resolvedUrl.startsWith(redirectionPathUrl)) {
+            return r.prefix + resolvedUrl.slice(redirectionPathUrl.length) as
+                PackageRelativeUrl;
+          }
+          return oldGetPackageRelativeUrl(resolvedUrl);
+        };
+    getPackageRelativeUrl = newGetPackageRelativeUrl;
+  });
+
   const loaders: UrlLoader[] = redirections.map(
-      (r: redirection) =>
-          new PrefixedUrlLoader(r.prefix, new FSUrlLoader(r.path)));
-  if (loaders.length > 0) {
+      (r: redirection) => new FSUrlLoader(path.resolve(r.path)));
+  if (redirections.length > 0) {
     options.analyzer = new Analyzer({
-      urlResolver:
-          new MultiUrlResolver([...resolvers, new PackageUrlResolver()]),
-      urlLoader:
-          new MultiUrlLoader([...loaders, new FSUrlLoader(projectRoot)])
+      urlResolver: new MultiUrlResolver([...resolvers, packageUrlResolver]),
```

Comment:
It doesn't technically get overwritten because it has no way to be set prior to here.  The options which are initialized in polymer-bundler.ts (which is the CLI) can not include an externally specified analyzer.